src/System-Platforms/OSPlatform.class.st

```diff
@@ -97,26 +97,34 @@ OSPlatform >> current [
 
 { #category : #accessing }
 OSPlatform >> currentWorkingDirectoryPath [
-	"This method calls the method getPwdViaFFI with arugement of a buffer size. By default it uses the defaultMaximumPathLength of each subclass as the buffer size."
-	^ self currentWorkingDirectoryPathWithBufferSize: self defaultMaximumPathLength
-
+	"We MUST decode raw bytes to support non-Ascii characters"
+	^ self currentWorkingDirectoryRawPath decodeWith: self environment defaultEncoding
```

Comment:
As @tesonep says, `environment` is an extension method, so this method is creating a circular dependency. (System-Platforms <-> System-OSEnvironments)